From d93eb3adcec90abfeb248915047912e9ddfd0445 Mon Sep 17 00:00:00 2001
From: Tom Kirkpatrick <tkp@kirkdesigns.co.uk>
Date: Fri, 26 Jul 2019 14:30:57 +0100
Subject: [PATCH] lnd: expose user agent name as config option

Expose the neutrino `UserAgentName` conffig option. This can be set by
starting lnd with the `--neutrino.useragentname=` flag.
---
 chainregistry.go | 1 +
 config.go        | 1 +
 docs/INSTALL.md  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/chainregistry.go b/chainregistry.go
index a209cc6628..e637c560c1 100644
--- a/chainregistry.go
+++ b/chainregistry.go
@@ -752,6 +752,7 @@ func initNeutrinoBackend(chainDir string) (*neutrino.ChainService, func(), error
 
 	neutrino.MaxPeers = 8
 	neutrino.BanDuration = time.Hour * 48
+	neutrino.UserAgentName = cfg.NeutrinoMode.UserAgentName
 
 	neutrinoCS, err := neutrino.NewChainService(config)
 	if err != nil {
diff --git a/config.go b/config.go
index c5efa55711..6691c0d8b0 100644
--- a/config.go
+++ b/config.go
@@ -181,6 +181,7 @@ type neutrinoConfig struct {
 	BanThreshold       uint32        `long:"banthreshold" description:"Maximum allowed ban score before disconnecting and banning misbehaving peers."`
 	FeeURL             string        `long:"feeurl" description:"Optional URL for fee estimation. If a URL is not specified, static fees will be used for estimation."`
 	AssertFilterHeader string        `long:"assertfilterheader" description:"Optional filter header in height:hash format to assert the state of neutrino's filter header chain on startup. If the assertion does not hold, then the filter header chain will be re-synced from the genesis block."`
+	UserAgentName      string        `long:"useragentname" description:"Used to help identify ourselves to other bitcoin peers"`
 }
 
 type btcdConfig struct {
diff --git a/docs/INSTALL.md b/docs/INSTALL.md
index 2141a6dd28..661223d468 100644
--- a/docs/INSTALL.md
+++ b/docs/INSTALL.md
@@ -193,6 +193,7 @@ neutrino:
       --neutrino.maxpeers=                                    Max number of inbound and outbound peers
       --neutrino.banduration=                                 How long to ban misbehaving peers.  Valid time units are {s, m, h}.  Minimum 1 second
       --neutrino.banthreshold=                                Maximum allowed ban score before disconnecting and banning misbehaving peers.
+      --neutrino.useragentname=                               Used to help identify ourselves to other bitcoin peers.
 ```
 
 ## Bitcoind Options
